pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -269,64 +240,13 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 
 	commonArgs = append(commonArgs, osdOnSDNFlag(c.Network, c.clusterInfo.CephVersion)...)
 
+	// RE-ADD OSD ON DIRECTORY?
+	// RE-ADD LEGACY OSD?
+
 	var command []string
 	var args []string
-	if !osd.IsDirectory && osd.IsFileStore && !osd.CephVolumeInitiated {
-		doConfigInit = false
-		// All scenarios except one can call the ceph-osd daemon directly. The one different scenario is when
-		// filestore is running on a device. Rook needs to mount the device, run the ceph-osd daemon, and then
-		// when the daemon exits, rook needs to unmount the device. Since rook needs to be in the container
-		// for this scenario, we will copy the binaries necessary to a mount, which will then be mounted
-		// to the daemon container.
-		doBinaryCopyInit = true
-		// Since the data path is mounted from a different source for filestore disks, don't chown it
-		doChownDataPath = false
-
-		command = []string{path.Join(k8sutil.BinariesMountPath, "tini")}
-		args = []string{
-			"--", path.Join(k8sutil.BinariesMountPath, "rook"),
-			"ceph", "osd", "filestore-device",
-			"--source-path", sourcePath,
-			"--mount-path", osd.DataPath,
-			"--",
-		}
-		args = append(args,
-			opspec.DaemonFlags(c.clusterInfo, osdID)...,
-		)
-		args = append(args,
-			"--foreground",
-			"--osd-data", osd.DataPath,
-			"--osd-uuid", osd.UUID,
-			"--osd-objectstore", storeType,
-			"--osd-max-object-name-len", "256",
-			"--osd-max-object-namespace-len", "64",
-			"--crush-location", fmt.Sprintf("root=default host=%s", osdProps.crushHostname),
-			// Set '--setuser-match-path' so that existing directory owned by root won't affect the daemon startup.
-			// For existing data store owned by root, the daemon will continue to run as root
-			"--setuser-match-path", osd.DataPath,
-		)
-
-		// If the Bluestore OSD was prepared with ceph-volume and not running on PVC
-	} else if osd.CephVolumeInitiated && osdProps.pvc.ClaimName == "" && !osd.IsFileStore {
```

Comment:
We still need this case for a bluestore OSD created with c-v, right? Or maybe it just simplified/moved somewhere else

Reply:
It's in the else :)